src/object_detection.py

```diff
@@ -0,0 +1,72 @@
+import numpy as np
+
+height = 10000              # vertical dimension in millimeters
+width = 20000               # horizontal dimension in millimeters
+cell_dim = 20               # dimension of cells in millimeters (cells are squares)
+
+cell_col = np.ceil(height/cell_dim)
+cell_row = np.ceil(width/cell_dim)
+
+# Tunable parameters to determine if something is an object.
+col_diff = 20           # Expected max difference between two adjacent cells in a column.
+counter_reps = 3        # Number of reps required to dictate it is an object.
+prev_diff = 1           # Amount of error needed to exceed for previous to be flags
+same_obj_diff = 20      # maximum diff between horizontal cells to be considered the same object
+
+
+def object_detection(points):
+
+    cells = np.zeros((cell_row, cell_col))
+
+    for tup in points:
+        # creates list with x, y, and z coordinate
+        x, y, z = tup
+        x = x//cell_col
+        y = y//cell_row
+
+        # Dictating the z value for the cell. Currently only finds the minimum value of the cell.
+        if x >= 0 and x < cell_col and y >= 0 and y < cell_row:
+            cells[x, y] = z if cells[x, y] == 0 else min(z, cells[x, y])
+
+    close_arr = np.zeros((1, cell_w))
+
+    for col in range(cell_col):
+        prev = 0        # Previous cell.
+        closest = 0     # Minimum z value for an object in the column.
+        counter = 0     # Counts the instances in which adjacent cells do not exceed diff.
+        min_obj = 0     # Minimum dist to detected object
+        for row in range(cell_row):
+            min_obj = cells[row, col] if counter = 0 else min(min_obj, cells[row, col])
+            if abs(cells[row, col] - prev) < col_diff:
```

Comment:
col_diff is defined as the `Expected max difference between two adjacent cells in a column.` This line implies that if the difference between 2 cells is below col_diff, that we are potentially looking at something different. Should the sign be swapped? I might just be thinking about it wrong.